src/client/visualizations/table/interactions/interaction-controller.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2017-2021 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { List } from "immutable";
+import { Datum, PseudoDatum } from "plywood";
+import * as React from "react";
+import { ReactNode } from "react";
+import { Clicker } from "../../../../common/models/clicker/clicker";
+import { Essence, VisStrategy } from "../../../../common/models/essence/essence";
+import { FilterClause } from "../../../../common/models/filter-clause/filter-clause";
+import { SeriesDerivation } from "../../../../common/models/series/concrete-series";
+import { Series } from "../../../../common/models/series/series";
+import { SeriesSort, SortDirection } from "../../../../common/models/sort/sort";
+import { Stage } from "../../../../common/models/stage/stage";
+import { Binary, Nullary, Ternary, Unary } from "../../../../common/utils/functional/functional";
+import { Fn } from "../../../../common/utils/general/general";
+import { ScrollerPart } from "../../../components/scroller/scroller";
+import { MEASURE_WIDTH, SEGMENT_WIDTH, SPACE_LEFT } from "../../../components/tabular-scroller/dimensions";
+import { measureColumnsCount } from "../../../components/tabular-scroller/utils/measure-columns-count";
+import { Highlight } from "../../highlight-controller/highlight";
+import { HoverElement, PositionHover, rowPosition, seriesPosition } from "../utils/calculate-hover-position";
+import { getFilterFromDatum } from "../utils/filter-for-datum";
+
+interface InteractionsProps {
+  onClick: Ternary<number, number, ScrollerPart, void>;
```

Comment:
Should name callbacks consistently. `onClick` vs `setSegmentWidth`. I think we should use domain names, not "onEvent" names